lib/ddtrace/contrib/rack/middlewares.rb

```diff
@@ -82,7 +82,18 @@ def call(env)
 
           # call the rest of the stack
           status, headers, response = @app.call(env)
-          [status, headers, response]
+
+          wrapped_response = response
+
+          if defined?(::Rack::BodyProxy)
+            wrapped_response = ::Rack::BodyProxy.new(response) do
+              finish(request_span, frontend_span, tracer, env, status, headers, response, original_env)
```

Comment:
When does this block run? Does it run for both sync and async requests?

Reply:
Yeah it should run for both. It runs when the response is closed. It should also run regardless of whether the response was successful or not. If you look in the Rack handlers you'll typically see a `body.close` or `response.finish` call in an ensure clause that triggers this block.